k2/csrc/array_test.cu

```diff
@@ -705,22 +696,16 @@ TEST(ArrayTest, TestArray1Io) {
 TEST(ArrayTest, TestArray2Io) {
   for (int i = 0; i < 20; i++) {
     ContextPtr c = GetCpuContext();
-    int32_t dim0 = RandInt(0, 4),
-        dim1 = RandInt(0, 4);
-    if (dim0 == 0)
-      dim1 = 0;
-    Array1<int32_t> a = RandUniformArray1<int32_t>(
-        c, dim0 * dim1, 0, 100);
+    int32_t dim0 = RandInt(1, 4), dim1 = RandInt(1, 4);
```

Comment:
I was trying to test arrays where one dim is zero, which I believe are allowed, no?  Not a super big deal but I like to test all cases.

Reply:
For now, we don't allow to create an empty Array2 from Array1, should we change that? Or I can just add a test for empty array, but not create from Array1.

https://github.com/k2-fsa/k2/blob/6acfe3d54140889a8456a4002cb393607c165c64/k2/csrc/array.h#L569-L575

Reply:
Updated, reverted the tests.